substdo.c

```diff
@@ -3,13 +3,9 @@
 #include "byte.h"
 #include "error.h"
 
-static int allwrite(op,fd,buf,len)
-register int (*op)();
-register int fd;
-register char *buf;
-register unsigned int len;
+static int allwrite(ssize_t (*op)(), int fd, char *buf, size_t len)
```

Comment:
Fuller `op` signature for symmetry with `oneread`?

Reply:
The other way round: I removed the full specification in oneread(), just because we just pass the pointers from substdio in, and that does not have the argument specification.